File: behaviorial/chain-of-responsability/typescript/base-handler.ts
import { Handler } from './interface';

export abstract class BaseHandler implements Handler {
	private nextHandler: Handler | null = null;

	public setNext(handler: Handler): Handler {
		this.nextHandler = handler;
		return handler;
	}

	public handle(request: string): string | null {
		if (this.nextHandler) {
			return this.nextHandler.handle(request);
		}
		return null;
	}
}

File: behaviorial/chain-of-responsability/python/base_handler.py
from typing import Optional

from interface import Handler


class BaseHandler(Handler):
    _next_handler: Optional[Handler] = None

    def set_next(self, handler: Handler) -> Handler:
        self._next_handler = handler
        return handler

    def handle(self, request: str) -> Optional[str]:
        if self._next_handler:
            return self._next_handler.handle(request)
        return None
